html.go

```diff
@@ -901,31 +902,10 @@ func doubleSpace(out *bytes.Buffer) {
 }
 
 func isRelativeLink(link []byte) (yes bool) {
-	// a tag begin with '#'
-	if link[0] == '#' {
-		return true
+	u, err := url.Parse(string(link[:]))
```

Comment:
You can convert directly from a byte slice to a string (note that in
general it has to make a copy of the contents to ensure that the resulting
string is immutable). The example you linked to uses a byte array. In that
case it is necessary to take a slice of the array, and then the slice can
be converted into a string.
On Jan 6, 2016 12:43 PM, "Gustavo Bittencourt" <notifications@github.com>
wrote:

> In html.go
> <https://github.com/russross/blackfriday/pull/231#discussion_r48999981>:
>
> > @@ -901,31 +902,10 @@ func doubleSpace(out *bytes.Buffer) {
> >  }
> >
> >  func isRelativeLink(link []byte) (yes bool) {
> > -	// a tag begin with '#'
> > -	if link[0] == '#' {
> > -		return true
> > +	u, err := url.Parse(string(link[:]))
>
> I got this hint in stackoverflow (details here
> <http://stackoverflow.com/questions/14230145/what-is-the-best-way-to-convert-byte-array-to-string>).
> The user ferhat elmas commented
> <http://stackoverflow.com/questions/14230145/what-is-the-best-way-to-convert-byte-array-to-string#comment55642294_18615786>
> about this issue:
>
> a string is in effect a read-only slice of bytes. You can't convert byte
> array directly to string so first slice then string".
>
> If you try a direct convert, you will get an error, like this example:
> http://play.golang.org/p/Wdr1_sOzMH
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/russross/blackfriday/pull/231/files#r48999981>.
>